packages/ffe-file-upload-react/src/FileUpload.js

```diff
@@ -29,97 +33,103 @@ class FileUpload extends React.Component {
         this.props.onFilesSelected(event.target.files);
     }
 
+    onFilesDropped(event) {
+        event.preventDefault();
+        this.setState({ hover: false });
+        this.props.onFilesDropped(event);
+    }
+
     onFileDeleted(event) {
-        this.props.onFileDeleted(
-            this.props.selectedFiles.find(
-                file => file.name === event.target.id,
-            ),
-        );
+        this.props.onFileDeleted(this.props.files[event.target.id]);
     }
 
     render() {
         const {
             id,
             label,
-            selectedFilesHeaderLabel,
-            selectedFiles,
-            accept,
+            files,
+            cancelText,
+            deleteText,
             multiple,
-            errorMessage,
-            infoMessage,
-            successMessage,
+            title,
+            infoText,
+            infoSubText,
+            uploadTitle,
+            uploadMicroText,
+            uploadSubText,
+            onFileDeleted,
         } = this.props;
 
         return (
             <div className="ffe-file-upload">
-                <button
-                    className="ffe-file-upload__button"
-                    aria-invalid={String(!!errorMessage)}
-                    onClick={this.triggerUploadFileNativeHandler}
+                <div className="ffe-file-upload__title">{title}</div>
+                {Object.keys(files).length > 0 ? (
+                    <div className="ffe-file-upload__file-items-section">
+                        {Object.keys(files).map((file, index) => (
+                            <FileItem
+                                key={index}
+                                index={index}
```

Comment:
`index` isn't needed here.
`key` is recommended to not use index from `.map()`, but rather some value which actually corresponds to the data you want to use. Perhaps simply using `file` here will do?

Edit: if `file` is the file name, we might get duplicates. You could try using `files[file]` instead.

Reply:
There's no issue with using the index as a key as long as you don't plan on changing the sort order.

Reply:
Hm.. well actually in this case that can happen since there is functionality to delete a file. If one is deleted from the front of the list then all the other ones will have their index changed so in this case it actually does matter.

I'll go to my corner.

Reply:
> `index` isn't needed here.
> `key` is recommended to not use index from `.map()`, but rather some value which actually corresponds to the data you want to use. Perhaps simply using `file` here will do?
> 
> Edit: if `file` is the file name, we might get duplicates. You could try using `files[file]` instead.

File is an object, so even if we have duplicates, they should not be equal, right?

Reply:
> File is an object, so even if we have duplicates, they should not be equal, right?

In the prop types comments here it says `files` is "A map of files, indexed by file name".
It occurs to me now that of course the `files` object can't have duplicate keys, but how does this data structure work if the user wants to upload two files with identical file names? :thinking: